python/lsst/pipe/base/executionButlerBuilder.py

```diff
@@ -141,14 +147,19 @@ def _setupNewButler(butler: Butler, outputLocation: ButlerURI, dirExists: bool)
     # file data stores continue to look at the old location.
     config = Config(butler._config)
     config["root"] = outputLocation.geturl()
+    config["allow_put_of_predefined_dataset"] = True
     config["registry", "db"] = "sqlite:///<butlerRoot>/gen3.sqlite3"
     # record the current root of the datastore if it is specified relative
     # to the butler root
     if config.get(("datastore", "root")) == BUTLER_ROOT_TAG:
         config["datastore", "root"] = butler._config.configDir.geturl()
     config["datastore", "trust_get_request"] = True
 
-    config = Butler.makeRepo(root=outputLocation, config=config, overwrite=True, forceConfigRoot=False)
+    # Requires that we use the dimension configuration from the original
+    # butler and not use the defaults.
+    config = Butler.makeRepo(root=outputLocation, config=config,
+                             dimensionConfig=butler.registry.dimensions.dimensionConfig,
```

Comment:
I wonder if this is the difference between how I was doing things before with initializing the registry explicity vs switching to `butler.makeRepo` to make the registry in terms of dimensions being transfered.

Reply:
We haven't changed dimensions config in a long time so I don't think this would have mattered. It was going to be a potential problem in the future though.